sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetReadState.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet;
+
+/**
+ * Helper class to store intermediate state while reading a Parquet column chunk.
+ */
+final class ParquetReadState {
+  /** Maximum definition level */
+  int maxDefinitionLevel;
+
+  /** The offset to add the next value in the current batch */
+  int offset;
+
+  /** The remaining number of values to read in the current page */
+  int valuesToReadInPage;
+
+  /** The remaining number of values to read in the current batch */
+  int valuesToReadInBatch;
```

Comment:
Just a question. Previously, we tracked with `long` type. After this PR, `int` is enough and this is aligned with Parquet file definition. Did I understand correctly?

Reply:
Yes. Previously `valuesRead` is the number of values read so far within the column chunk, so it has to be `long`. However, it is not that useful. Here we changed to only track values read so far within a page and within a single batch, both of which are integers.